### ISSUE TITLE
Missing regexTypes.d.ts in distributed module

### ISSUE THREAD

**Issue by svvac** [sql-formatter-org/sql-formatter#353]
There seems to be a problem with the distribution of the [`regexTypes.d.ts`](https://github.com/sql-formatter-org/sql-formatter/blob/master/src/lexer/regexTypes.d.ts) file in the module available on npm.

When typescripts checks a project using the library, it complains:

```
node_modules/sql-formatter/lib/src/lexer/Tokenizer.d.ts:2:29 - error TS2307: Cannot find module './regexTypes' or its corresponding type declarations.

2 import * as regexTypes from "./regexTypes";
                              ~~~~~~~~~~~~~~


Found 1 error in node_modules/sql-formatter/lib/src/lexer/Tokenizer.d.ts:2
```

`lib/src/lexer` indeed does not contain the requisite file. It appears that the compilation step considers it as a plain typescript file and emits an (empty) module in `lib/lexer` instead:

```console
$ find node_modules/sql-formatter -name 'regexTypes.*'
node_modules/sql-formatter/lib/lexer/regexTypes.d.js
node_modules/sql-formatter/lib/lexer/regexTypes.d.js.map
```

The issue appears in (at least) 8.2 and 8.1.

**Comment by svvac** [comment]
Quick workaround, for those using yarn 2+ and its patch feature:

Add the following resolution to your `package.json` (note that the version specifier in resolutions and dependencies [must match _exactly_][1]):

```patch
 {
   "dependencies": {
     "sql-formatter": "^8.2.0"
   },
   "resolutions": {
+    "sql-formatter@^8.2.0": "patch:sql-formatter@npm:8.2.0#.yarn/patches/sql-formatter-npm-8.2.0-cee6286818.patch"
   }
}
```

[1]: https://github.com/yarnpkg/berry/issues/4231#issuecomment-1072868948

<details>
<summary>Patch file <code>.yarn/patches/sql-formatter-npm-8.2.0-cee6286818.patch</code></summary>

```patch
diff --git a/lib/src/lexer/regexTypes.d.ts b/lib/src/lexer/regexTypes.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..71fa1306c6bb19eef9745397999eac3a8a34b43a
--- /dev/null
+++ b/lib/src/lexer/regexTypes.d.ts
@@ -0,0 +1,29 @@
+import { quotePatterns } from './regexFactory';
+
+export interface IdentChars {
+  // Additional characters that can be used as first character of an identifier.
+  // That is: in addition to letters and underscore.
+  first?: string;
+  // Additional characters that can appear after the first character of identifier.
+  // That is: in addition to letters, numbers and underscore.
+  rest?: string;
+  // True to allow single dashes (-) inside identifiers, but not at the beginning or end
+  dashes?: boolean;
+}
+
+export type PlainQuoteType = keyof typeof quotePatterns;
+
+export type PrefixedQuoteType = {
+  quote: PlainQuoteType;
+  prefixes: string[];
+  requirePrefix?: boolean; // True when prefix is required
+};
+
+export type QuoteType = PlainQuoteType | PrefixedQuoteType;
+
+export interface VariableRegex {
+  regex: string;
+}
+
+export type VariableType = VariableRegex | PrefixedQuoteType;
+
```

</details>

**Comment by nene** [comment]
Fix for this is now released in 9.0.0-beta.6
